codec-redis/src/main/java/io/netty/handler/codec/redis/RedisConstants.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.netty.handler.codec.redis;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.util.internal.PlatformDependent;
+
+/**
+ * Constant values for Redis encoder/decoder.
+ */
+final class RedisConstants {
+
+    private RedisConstants() {
+    }
+
+    static final int TYPE_LENGTH = 1;
+
+    static final int EOL_LENGTH = 2;
+
+    static final int NULL_LENGTH = 2;
+
+    static final int NULL_RESPONSE_LENGTH = -1;
+
+    static final int REDIS_MESSAGE_MAX_LENGTH = 512 * 1024 * 1024; // 512MB
+
+    static final int POSITIVE_LONG_MAX_LENGTH = 19; // length of Long.MAX_VALUE
+
+    static final int LONG_MAX_LENGTH = POSITIVE_LONG_MAX_LENGTH + 1; // +1 is sign
+
+    static final short NULL_SHORT = makeShort('-', '1');
+
+    static final short EOL_SHORT = makeShort('\r', '\n');
+
+    static ByteBuf nullByteBuf() {
+        return NULL_BYTEBUF.slice();
```

Comment:
I think I may have suggested making these ByteBufs for demonstration purposes at one point and confused things ... I think we should make these `byte[]` to avoid the relatively high overhead of slicing small fixed content ByteBuf objects.